vcpkg/maintainers/functions/vcpkg_make_configure.md

```diff
@@ -0,0 +1,162 @@
+
+Setting `DISABLE_CPPFLAGS` disables that behavior, leaving preprocessor relevant flags in `CFLAGS` and `CXXFLAGS`.
+
+This can be useful for projects that do not use standard environment variables or have specific preprocessor requirements.
+
+### DISABLE_DEFAULT_OPTIONS
+
+Setting this flag disables the following default configure options provided by `vcpkg_make_configure`. Use this flag if you need full control over the configure command line.
+
+General Defaults:
+* --disable-silent-rules - Makes the build process output more verbose.
+* --verbose - Enables verbose output from the configure script.
+
+Library Linkage:
+* For dynamic library builds: --enable-shared and --disable-static
+* For static library builds: --disable-shared and --enable-static
+
+Installation Directories:
+* Adjusts standard directories (bin, lib, share, etc.) to conform with vcpkg's layout, ensuring correct installation paths.
+
+Platform-Specific Tweaks:
+* On Windows: Ensures compatibility with Windows-specific filesystem and library behaviors.
+* On Unix-like systems: Assures the script accommodates the typical filesystem layout and toolchain behaviors.
+
+Removing these defaults gives you the responsibility to specify all necessary configuration options manually, ensuring the project is correctly configured for building with vcpkg.
+
+### USE_RESPONSE_FILES
+
+Enables the use of response files to pass arguments to the linker.
+
+If set, the code generates response files for the linker flags (`CPPFLAGS`, `CFLAGS`, `CXXFLAGS`, `LDFLAGS`) and passes the response file path to the linker instead of the individual flags. If it's false, the flags are passed directly.
+
+This can be helpful when dealing with long command lines that exceed the maximum allowed length.
+
+### SOURCE_PATH
+
+The directory containing the project's source files
+
+This value is usually obtained as a result of calling a source acquisition command like [`vcpkg_from_github()`](vcpkg_from_github.md).
+
+### OPTIONS
+
+Additional options to pass to the configure script. Use these to specify any custom flags or settings required by the project.
+
+### OPTIONS_RELEASE
+
+Additional options to pass to the configure script. Use these to specify any custom flags or settings required by the project.
+
+These are in addition to `OPTIONS`.
+
+### OPTIONS_DEBUG
+
+Additional options to pass to the configure script. Use these to specify any custom flags or settings required by the project.
+
+These are in addition to `OPTIONS`.
+
+### PRE_CONFIGURE_CMAKE_COMMANDS
+
+Specifies a list of CMake commands to execute before the configure process for each build type. That is, if `VCPKG_BUILD_TYPE` is empty (the default), these commands will run once for debug and once for release. If `VCPKG_BUILD_TYPE` is `release`, these commands will only be once for release. This can be used to set up the environment or prepare the build in a way that the configure script requires.
```

Comment:
This needs an example. Do I pass the name of a function? What arguments does it receive?

```cmake
function(my_preconfigure_command SRC BUILD_TYPE)
    # Each source directory gets different version.txt files
    if(BUILD_TYPE STREQUAL "release")
        file(WRITE "${SRC}/version.txt" "This is a release version!")
    else()
        file(WRITE "${SRC}/version.txt" "This is a debug version!")
    endif()
    # The environment variable has different values for each configuration
    set(ENV{SPECIAL} "${BUILD_TYPE} is magic")
endfunction()

vcpkg_make_configure(
    # ...
    PRE_CONFIGURE_CMAKE_COMMANDS my_preconfigure_command
)
```